astronomer/providers/amazon/aws/operators/sagemaker.py

```diff
@@ -91,3 +100,188 @@ def execute_complete(self, context: Context, event: Any = None) -> Dict[str, Any
+    After training completes, Amazon SageMaker saves the resulting model artifacts to an Amazon S3 location
+    that you specify.
+
+    .. seealso::
+        For more information on how to use this operator, take a look at the guide:
+        :ref:``howto/operator:SageMakerTrainingOperator``
+
+    :param config: The configuration necessary to start a training job (templated).
+        For details of the configuration parameter see ``SageMaker.Client.create_training_job``
+    :param aws_conn_id: The AWS connection ID to use.
+    :param print_log: if the operator should print the cloudwatch log during training
+    :param check_interval: if wait is set to be true, this is the time interval
+        in seconds which the operator will check the status of the training job
+    :param max_ingestion_time: The operation fails if the training job
+        doesn't finish within max_ingestion_time seconds. If you set this parameter to None,
+        the operation does not timeout.
+    :param check_if_job_exists: If set to true, then the operator will check whether a training job
+        already exists for the name in the config.
+    :param action_if_job_exists: Behaviour if the job name already exists. Possible options are "increment"
+        (default) and "fail".
+        This is only relevant if check_if_job_exists is True.
+    """
+
+    def execute(self, context: Context) -> None:  # type: ignore[override]
+        """
+        Creates SageMaker training job via sync hook `create_training_job` and pass the
+        control to trigger and polls for the status of the training job in async
+        """
+        self.preprocess_config()
+        if self.check_if_job_exists:
+            self._check_if_job_exists()
+        self.log.info("Creating SageMaker training job %s.", self.config["TrainingJobName"])
+        response = self.hook.create_training_job(
+            self.config,
+            wait_for_completion=False,
+            print_log=False,
+            check_interval=self.check_interval,
+            max_ingestion_time=self.max_ingestion_time,
+        )
+        if response["ResponseMetadata"]["HTTPStatusCode"] != 200:
+            raise AirflowException(f"Sagemaker Training Job creation failed: {response}")
+        else:
+            end_time: Optional[float] = None
+            if self.max_ingestion_time is not None:
+                end_time = time.time() + self.max_ingestion_time
+            if self.print_log:
+                description = self.hook.describe_training_job(self.config["TrainingJobName"])
+                self.log.info(secondary_training_status_message(description, None))
+                instance_count = description["ResourceConfig"]["InstanceCount"]
+                status = description["TrainingJobStatus"]
+                self.defer(
+                    timeout=self.execution_timeout,
+                    trigger=SagemakerTrainingWithLogTrigger(
```

Comment:
SagemakerTrainingWithLogTrigger "with log" is this a customer ask? look like it making the operator a bit complicated and the log will be available on the trigger only so that the log at will be not that visible on astro.  cc: @phanikumv @kaxil

Reply:
thank you for clarifying it. Right, I was checking the sync version it does the same also I see a couple of methods in the PR that are almost the same as the sync version just think if we can just that function to convert to async than re-writing all logic. Maybe we can as separate PR. that way we have to write less test case and also it would be relatively easy to manage

Reply:
test cover for each line of these functions is also time taking.